### PR TITLE
SAK-29520 : Upgrade copy site

### PR DESCRIPTION
**Pull request by frasese**
Tools affected :
- Announcements
- Assignments
- Calendar
- Forums
- Syllabus
- Polls
- Wiki
- Kernel*
- Site-manage*

### CODE REVIEW

**Comment by buckett** [comment]
This needs an update to Dashboard to allow it to build.

**Comment by buckett** [comment]
Doesn't appear to build because of broken mocks.
Also doesn't merge.

**Comment by frasese** [comment]
Rebased...
Dashboard tests still broken, but they have nothing to do with this patch.

File: kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/ContentCopyImpl.java [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+				newCollection.setAvailability(resource.isHidden(), resource.getReleaseDate(), resource.getRetractDate());
+				hideImportedContent(newCollection);
+
+				// Copy the permissions across
+				try{
+					AuthzGroup oldRealm = ags.getAuthzGroup(resource.getReference());
+					ags.addAuthzGroup(newCollection.getReference(), oldRealm, null);
+				} catch (GroupNotDefinedException e) {
+					// do nothing - this case is expected to be common
+				} catch (GroupAlreadyDefinedException e) {
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/ContentCopyImpl.java]: Group Exceptions with the same actions.

File: kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java [hunk 2]
```diff
@@ -8049,283 +8065,71 @@ private String siteIdSplice(String ref, String siteId){
...
-							}
-							catch (IdInvalidException e)
-							{
-							}
-							catch (InconsistentException e)
-							{
-							}
-							catch (ServerOverloadException e)
-							{
-							}
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java]: Nice cleaning up these useless catch blocks.

File: kernel/kernel-impl/src/test/java/org/sakaiproject/content/impl/ContentCopyIntTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+			}
+
+			protected void tearDown() throws Exception {
+				oneTimeTearDown();
+			}
+		};
+		return setup;
+	}
+
+	public void testSimpleCopy() throws Exception {
```
Comment (ern) [inline kernel/kernel-impl/src/test/java/org/sakaiproject/content/impl/ContentCopyIntTest.java]: Thanks for creating unit test

**Comment by buckett** [comment]
The dashboard failure is related to a mock that's not been updated.

**Comment by frasese** [comment]
Rebased with master (12x)

**Comment by steveswinsburg** [comment]
This is still in conflict. Whats the status of this, are we agreeable for this going in once its fixed? It's been open for almost a year.

**Comment by buckett** [comment]
I've rebase this and created #4404

**Pull request closed without merging**